crates/gpui/src/window.rs

```diff
@@ -261,15 +261,20 @@ impl FocusHandle {
     pub(crate) fn for_id(id: FocusId, handles: &Arc<FocusMap>) -> Option<Self> {
         let lock = handles.read();
         let ref_count = lock.get(id)?;
-        if ref_count.load(SeqCst) == 0 {
-            None
-        } else {
-            ref_count.fetch_add(1, SeqCst);
-            Some(Self {
-                id,
-                handles: handles.clone(),
-            })
+        let mut loaded = ref_count.load(SeqCst);
+        loop {
+            if loaded == 0 {
```

Comment:
Rather than duplicating this logic, can we move it into a function?

Reply:
fixed